pkg/kubelet/eviction/helpers.go

```diff
@@ -1254,7 +1254,11 @@ func evictionMessage(resourceToReclaim v1.ResourceName, pod *v1.Pod, stats stats
 				requests := container.Resources.Requests[resourceToReclaim]
 				if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) &&
 					(resourceToReclaim == v1.ResourceMemory || resourceToReclaim == v1.ResourceCPU) {
-					if cs, ok := podutil.GetContainerStatus(pod.Status.ContainerStatuses, container.Name); ok {
+					containerStatuses := pod.Status.ContainerStatuses
+					if utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) && kubetypes.IsRestartableInitContainer(&container) {
```

Comment:
I'm not sure I like the idea of using `IsRestartableInitContainer` when you're not sure if you're handling an init container. It assumes that RestartPolicy will never apply to non-init containers. I think the better approach is the one you've used elsewhere: just aggregate the ContainerStatuses & InitContainerStatuses together, and don't worry about whether the container in question is an init container.

Reply:
Thanks for the suggestion. Applied it.